phpcs/Standards/QualityAssurance/Sniffs/Generic/HardcodedPathSniff.php

```diff
@@ -59,25 +59,28 @@ public function process(PHP_CodeSniffer_File $phpcsFile, $stackPtr)
 
         // Path regular expression.
         $regexp = 'sites/[^/]+/(files|libraries|modules|themes)';
-
         // If hardcoded path is found.
         if (preg_match("~$regexp~", $token['content'], $matches)) {
-            $error = "Internal hardcoded paths are not allowed. ";
-            switch ($matches[1]) {
-                case 'modules':
-                    $error .= "Please use drupal_get_path('module', \$name).";
-                    break;
-                case 'themes':
-                    $error .= "Please use drupal_get_path('theme', \$name).";
-                    break;
-                case 'libraries':
-                    $error .= "Please use libraries_get_path(\$name).";
-                    break;
-                case 'files':
-                    $error .= "Please use variable_get('file_public|private|temporary_path').";
-                    break;
+            // Check if it's not part of 'variable_get'.
+            $findPrevious = $phpcsFile->findPrevious(T_STRING, $stackPtr);
+            if ($tokens[$findPrevious]['content'] !== 'variable_get') {
```

Comment:
Hi @zarabatana , for some reason the sniff is not detecting the previous examples like:
```
$imgUrl = 'sites/default/files/my_image.png';
```